File: emcrypt-core/src/main/java/com/beam/emcryptcore/model/auth/Account.java
package com.beam.emcryptcore.model.auth;

import com.beam.emcryptcore.base.Base;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;
import org.springframework.data.annotation.TypeAlias;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

import static java.util.Collections.emptyList;

@Document("Account")
@TypeAlias("Account")
@AllArgsConstructor
@NoArgsConstructor
@Accessors(chain = true)
@Data
@Builder
public class Account extends Base {

    private String username;
    private String password;

    private List<Role> authorities;
    private Profile profile;

    @Override
    public Account newIdAndCreated() {
        profile = Profile.init();

        return super.newIdAndCreated();
    }

    private boolean accountNonExpired = true;
    private boolean accountNonLocked = true;
    private boolean credentialsNonExpired = true;
    private boolean enabled = true;
}

File: emcrypt-web/src/main/ui/src/app/modules/auth/models/account.ts
export class Role {
  name: string;
}

export class Account {
  id: string;
  username: string;
  password: string;
  authorities: Role[];
  accountNonExpired: boolean;
  accountNonLocked: boolean;
  credentialsNonExpired: boolean;
  enabled: boolean;

  setUser(_user: Account) {
    this.id = _user.id;
    this.username = _user.username || '';
    this.password = _user.password || '';
    this.authorities = _user.authorities || [];
    this.accountNonExpired = _user.accountNonExpired || true;
    this.accountNonLocked = _user.accountNonLocked || true;
    this.credentialsNonExpired = _user.credentialsNonExpired || true;
    this.enabled = _user.enabled || true;
  }

  hasRole(role: string): boolean {
    let has = false;
    this.authorities.forEach((a) => {
      if (a.name === role) {
        has = true;
      }
    });
    return has;
  }
}

export class AuthResponse {
  code: number;
  token: string;
  account: Account;

  setAuth(auth: AuthResponse) {
    this.code = auth.code;
    this.token = auth.token;
    this.account = auth.account;
  }
}
